src/visualization/components/annotationController.tsx

```diff
@@ -0,0 +1,143 @@
+// React
+import {Dispatch} from 'react'
+
+// Redux
+import {writeThenFetchAndSetAnnotations} from 'src/annotations/actions/thunks'
+
+import {AnnotationsList} from 'src/types'
+
+import {showOverlay, dismissOverlay} from 'src/overlays/actions/overlays'
+import {event} from 'src/cloud/utils/reporting'
+import {notify} from 'src/shared/actions/notifications'
+import {createAnnotationFailed} from 'src/shared/copy/notifications'
+import {getErrorMessage} from 'src/utils/api'
+
+import {
+  AnnotationLayerConfig,
+  InfluxColors,
+  InteractionHandlerArguments,
+} from '@influxdata/giraffe'
+
+export const makeAnnotationClickListener = (
+  dispatch: Dispatch<any>,
```

Comment:
this method only dispatches two possible things, right? the output of `writeThenFetchAndSetAnnotations` and the output of `notify(createAnnotationFailed(getErrorMessage(err)))`

is it possible to template this dispatch call to those two return types?

something like
```ts
import {Action as NotificationAction} from 'src/shared/actions/notifications'
const SuccessAction = ReturnType<typeof setAnnotations>

Dispatch<SuccessAction | NotificationAction>
``` 

?

Reply:
let's talk about this in person, I don't have enough context (i need to learn more about dispatch and actions)

Reply:
conclusion:  we are using <any> because it is non-trivial and wonky.